docs/application/native/guides/device/battery-monitor.md

```diff
@@ -79,90 +77,105 @@ To get the battery usage information related to the application ID:
+    battery_monitor_resource_id_e resource_id = BATTERY_MONITOR_RESOURCE_ID_CPU;
 
-    error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage);
+    error_code = battery_monitor_usage_data_get_power_usage_per_resource(data_handle, resource_id, &battery_usage);
     if (error_code == BATTERY_MONITOR_ERROR_NONE)
-        printf("Battery Usage by display is [%d]", battery_usage);
+        printf("Battery Usage by CPU is [%lf]", battery_usage);
 
     resource_id = BATTERY_MONITOR_RESOURCE_ID_WIFI;
 
-    error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage)
+    error_code = battery_monitor_usage_data_get_power_usage_per_resource(data_handle, resource_id, &battery_usage);
     if (error_code == BATTERY_MONITOR_ERROR_NONE)
-        printf("Battery Usage by display is [%d]", battery_usage);
+        printf("Battery Usage by wifi is [%lf]", battery_usage);
+    else if (error_code == BATTERY_MONITOR_ERROR_RECORD_NOT_FOUND)
+        printf("Battery usage for above resource is not recorded");
 
-    /* Similarly for other resources...
+    /* Similarly for other resources.
     .
+    . You can also iterate over the resource enums using 'BATTERY_MONITOR_RESOURCE_ID_MAX'
     .
     .
     */
 
-    /*To avoid memory leak free the handle*/
-    error_code = battery_monitor_destroy(data_handle);
+    /*To avoid memory leak, free the handle*/
+    error_code = battery_monitor_battery_usage_data_destroy(data_handle);
     if (error_code != BATTERY_MONITOR_ERROR_NONE)
         printf("Error Occurred [%d]", error_code);
     ```
 
 <a name="resourceusage_get"></a>
 ## Getting Battery Usage Information of Resource
 
-To get battery usage information related to the resource ID:
+To get the battery usage information related to the resource ID:
 
-- Get the battery usage information of a particular resource over a specific duration of time.
-  To get the resource ID, use the `battery_monitor_resource_id_e` enum and to get time duration type information, use the `battery_monitor_duration_type_e` enum:
+- Get the battery usage information of a particular resource over a specific interval of time.
+  To get the resource ID, use the `battery_monitor_resource_id_e` enum. The timestamps **from** and **to** must be set as per the Unix Epoch time format:
 
     ```
     resource_id = BATTERY_MONITOR_RESOURCE_ID_DISPLAY;
-    error_code = battery_monitor_get_total_usage_by_resource_id(resource_id, duration_val, &battery_usage);
+    error_code = battery_monitor_get_power_usage_by_resource(resource_id, from, to, &battery_usage);
     if (error_code == BATTERY_MONITOR_ERROR_NONE)
-        printf("Battery Usage Display in last 24 hrs is [%d]", battery_usage);
+        printf("Battery Usage Display in last 1000 seconds is [%lf]", battery_usage);
+    else if (error_code == BATTERY_MONITOR_ERROR_RECORD_NOT_FOUND)
+        printf("Battery Usage for Display not recorded");
```

Comment:
@nawab-samsung Question. According to your other comments, isn't "by" better than "for"?

Battery Usage _by_ Display ...

Reply:
@d1-
```suggestion
 printf("The Battery Usage by Display is not recorded");